pkg/divert/istio/divert.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istio
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/okteto/okteto/pkg/k8s/virtualservices"
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/model"
+	istioV1beta1 "istio.io/client-go/pkg/apis/networking/v1beta1"
+	istioclientset "istio.io/client-go/pkg/clientset/versioned"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	UPDATE_CONFLICT_RETRIES = 10
+)
+
+// Driver weaver struct for the divert driver
+type Driver struct {
+	Client      kubernetes.Interface
+	IstioClient istioclientset.Interface
+	Manifest    *model.Manifest
+	cache       *divertIstioCache
+}
+
+type divertIstioCache struct {
+	divertVirtualServices    map[string]*istioV1beta1.VirtualService
+	developerVirtualServices map[string]*istioV1beta1.VirtualService
+}
```

Comment:
Can we rename this to
```suggestion
type virtualServicesCache struct {
	divert    map[string]*istioV1beta1.VirtualService
	developer map[string]*istioV1beta1.VirtualService
}
```
? The most important distinction along the code is if it's a `divert` or a `developer` VS and it's hard to differiantiate if they have both the `VirtualSevices` suffix